src/Xrefcheck/System.hs

```diff
@@ -36,21 +46,118 @@ readingSystem = unsafePerformIO
 -- Check the respective env variable which is usually set in all CIs.
 askWithinCI :: IO Bool
 askWithinCI = lookupEnv "CI" <&> \case
-  Just "1"                       -> True
+  Just "1" -> True
   Just (map C.toLower -> "true") -> True
-  _                              -> False
+  _ -> False
+
+-- | A FilePath that has been canonicalized. Should be created via
+-- 'canonicalizePath'. Currently, canonical paths have been made absolute,
+-- normalised regarding the running platform (e.g. Posix or Windows), with
+-- indirections syntractically expanded as much as possible and with no
+-- trailing path separator. All this results in a weaker version than that
+-- provided by 'System.Directory.canonicalizePath'.
```

Comment:
Good explanation :+1: 

On the formatting: it seems common to put a very brief explanation at the first line, and then elaborate at at a separate line. Sentences about different moments are also better put on separate lines.

Note, that to make two sentences appear at separate lines in the rendered Haddock documentation, you have to separate those with a blank line in the code, like:

```hs
-- A FilePath that has been canonicalized.
--
-- Should be created via 'canonicalizePath'.
```